std/machines/arith16.asm

```diff
@@ -0,0 +1,191 @@
+use std::array;
+use std::utils::unchanged_until;
+use std::utils::force_bool;
+use std::utils::sum;
+use std::math::ff;
+use std::check::panic;
+use std::convert::int;
+use std::convert::fe;
+use std::convert::expr;
+use std::prover::eval;
+use std::prelude::Query;
+use std::machines::range::Byte;
+
+// Arithmetic machine, ported mainly from Polygon: https://github.com/0xPolygonHermez/zkevm-proverjs/blob/main/pil/arith.pil
+// This machine supports eq0, which is the affine equation. Currently we only expose operations for mul and div.
+machine Arith16 with
+    latch: CLK8_7,
+    operation_id: operation_id,
+    // Allow this machine to be connected via a permutation
+    call_selectors: sel,
+{
+    Byte byte;
+    
+    col witness operation_id;
+
+    // operation_id has to be either mul or div.
+    force_bool(operation_id);
+
+    // Computes x1 * y1 + x2, where all inputs / outputs are 32-bit words (represented as 16-bit limbs in little-endian order).
+    // More precisely, affine_256(x1, y1, x2) = (y2, y3), where x1 * y1 + x2 = 2**16 * y2 + y3
+
+    // x1 * y1 = y2 * 2**16 + y3
+    operation mul<0> x1c[0], x1c[1], y1c[0], y1c[1] -> y2c[0], y2c[1], y3c[0], y3c[1];
```

Comment:
small detail, but can we switch to big endian here? I think it's more natural to represent numbers (known width) and the other machines are also written like that.

So it would be

```
mul A_high, A_low, B_high, B_low, -> result_high_high, result_high_low, result_low_high, result_low_low
div A_high, A_low, B_high, B_low, -> quot_high, quot_low, rem_high, rem_low
```

Reply:
Done!